### PR TITLE
CNI Project Security Self-Assessment - Security Pals

### PR DESCRIPTION
**Pull request by jasonliny**
Created and added first draft for CNI Project Security Self-Assessment.
Please feel free to share your feedback on the security self-assessment!

### CODE REVIEW

File: assessments/projects/cni/docs/CNI_SBOM_cyclonedx.bom.json
Comment (eddie-knight) [inline assessments/projects/cni/docs/CNI_SBOM_cyclonedx.bom.json:1]: I noticed that you included an SBOM along with the self assessment. There are two reasons that jump to the front of my mind for why this isn't needed...

1. SBOMs should be associated with releases, as the bill of materials is only accurate and useful if it is created at build time and associated to a particular point in the code history.
1. If you need to link to an SBOM for some reason in the self-assessment, you can just provide a link out to the latest build artifacts that contain an SBOM.

We still have plenty more to review, but as a starter— could you please remove the SBOM from this PR?

File: assessments/projects/cni/self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
+# Self-assessment
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:1]:
```suggestion
# CNI Self-assessment
```

File: assessments/projects/cni/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+The Self-assessment is the initial document for projects to begin thinking about the
+security of the project, determining gaps in their security, and preparing any security
+documentation for their users. This document is ideal for projects currently in the
+CNCF **sandbox** as well as projects that are looking to receive a joint assessment and
+currently in CNCF **incubation**.
+
+# Self-assessment outline
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:8]:
```suggestion
```

Please go back to ensure that no template text like this is included here or in the rest of the document

**Comment by jasonliny** [comment]
Thank you for the initial feedback. 
I have removed the SBOM for the time being since CNI doesn't generate one at the moment and I believe I have removed any preexisting template text.

File: assessments/projects/cni/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+### Conclusion
+
+Ongoing evaluation and updates to this model are necessary to adapt to evolving security threats in the CNI project.
+
+## Action Items
+
+- Consider creation of an SBOM for CNI
+- Security policy: Include what response can be expected after a vulnerability has been reported and consider creating a file or page dedicated to the vulnerability reporting process. ([Example Security Policy Template](https://github.com/cncf/tag-security/blob/main/project-resources/templates/SECURITY.md)).
+- Incident Response: Create a publicly available document outlining how security issues are triaged, how any parties are notified, and details surrounding the publication and release of a patch. ([Example Incident Response Template](https://github.com/cncf/tag-security/blob/main/project-resources/templates/incident-response.md))
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:277]: Following discussions from the review team, we are asking that Security Pals please remove any recommendations from the scope of this document. You may optionally communicate recommendations for projects in a more accessible location, such as in a project community channel or GitHub Issue.

File: assessments/projects/cni/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+* Dynamic policies for network bandwidth and firewall rules
+
+**Security**
+* Stop a container runtime from executing CNI plugins
+* Restrict what a plugin does
+* Manage the security of the container it interacts with
+
+## Self-assessment use
+
+This self-assessment is created by the CNI team to perform an internal analysis of the
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:125]: Was this created with active participation and review by the CNI team? If not, please remove that reference or replace it with a more accurate statement.

Consider listing the assessment contributors in the topmost paragraph, below the title and before the Table of Contents.

**Comment by jasonliny** [comment]
I removed the recommendations, added the authors near the top, and clarified that the self-assessment was made for the CNI project as opposed to being created by the CNI team.

File: assessments/projects/cni/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+| Doc | url |
+| -- | -- |
+| Security file | [README.md#Security](https://github.com/containernetworking/cni/blob/main/README.md#Security) |
+| Default and optional configs | [CNI Specification](https://www.cni.dev/docs/spec/) outlines required keys, optional keys, and protocol parameters |
+
+## Overview
+
+**Container Network Interface** <br>
+CNI, a project under the Cloud Native Computing Foundation, includes a set of guidelines and software libraries
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:43]:
```suggestion
Container Network Interface (CNI), a project under the Cloud Native Computing Foundation, includes a set of guidelines and software libraries
```

File: assessments/projects/cni/self-assessment.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+used for developing plugins that set up network interfaces within Linux and Windows containers. Its primary goal
+is managing container network connectivity and deallocating resources once a container is removed. This specific
+focus has led to broad support for CNI and a straightforward implementation process due to its simple specification.
+
+### Background
+
+With the rapid development of application containers on Linux, networking in this area is still not well addressed
+as it is highly environment-specific. To deal with this problem, developers seek to create a network layer that is
+pluggable for container runtimes. Ultimately, CNI, along with libraries for Go and a set of plugins, was introduced
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:52]:
```suggestion
pluggable for container runtimes. Ultimately, CNI was introduced
```
Consider moving `along with libraries for Go and a set of plugins` to a sentence that can elaborate more specifically on what libraries and plugins you're referencing here.

**Comment by jasonliny** [comment]
I have made the corrections to the first sentence in the Overview and in the Background section, added a sentence expanding on how a library is provided for integrating CNI into products and that the team maintains some reference plugins.

File: assessments/projects/cni/self-assessment.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+
+* Category: Denial of Service, Tampering
+* Certain configurations may lead to unexpected denial of service or tampering due to a plugin's precedence rules or if a configuration is invalid.
+* Possible Mitigation Strategies:
+    * Provide clear documentation on secure configuration practices.
+    * Implement configuration validation tools.
+
+#### Conclusion
+
+Ongoing evaluation and updates to this model are necessary to adapt to evolving security threats in the CNI project.
```
Comment (eddie-knight) [inline assessments/projects/cni/self-assessment.md:273]: Following discussions among the review team, we are asking that all STRIDE content be moved to a secondary document within this directory. Please extract this content and add a link to the new file in the appendix.

Comment (jasonliny) [reply assessments/projects/cni/self-assessment.md]: Got it. It has been moved to a separate document and a link to it is provided.

File: assessments/projects/cni/docs/threat-modeling.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# CNI Threat Modeling
```
Comment (torinvdb) [inline assessments/projects/cni/docs/threat-modeling.md:1]: Minor suggestion: consider renaming this to something like threat analysis, threat enumeration, lightweight threat assessment, etc. Presenting it as a threat model might be a bit misleading as that entails a comprehensive process and would be out of scope for the self-assessment (see: [OWASP Threat Modeling Process](https://owasp.org/www-community/Threat_Modeling_Process)).

Comment (jasonliny) [reply assessments/projects/cni/docs/threat-modeling.md]: Thanks for the suggestion. I wasn't aware of the differences in terminology so I really appreciate the feedback!
I've renamed the references of threat modeling to lightweight threat assessment so it's clearer in describing the work we've done.

Comment (torinvdb) [reply assessments/projects/cni/docs/threat-modeling.md]: No worries, looks good overall - thank you!

File: assessments/projects/cni/self-assessment.md [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+#### Action 4: Plugin Delegation
+
+There are some operations that, for whatever reason, cannot reasonably be implemented as a discrete chained plugin. Rather, a CNI plugin may wish to delegate some functionality to another plugin. One common example of this is IP address management.
+
+#### Action 5: Result Types
+
+Plugins can return one of three result types:
+* Success
+* Error
+* \_Version
```
Comment (JustinCappos) [inline assessments/projects/cni/self-assessment.md:104]: I wish we had more context here about why these are listed and what the meaning of these is in context...

File: assessments/projects/cni/docs/lightweight-threat-assessment.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# CNI Lightweight Threat Assessment
```
Comment (JustinCappos) [inline assessments/projects/cni/docs/lightweight-threat-assessment.md:1]: Why is this file included?  (I'm not saying to remove it, but I want to understand why we're adding it as well.)

File: assessments/projects/cni/self-assessment.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+
+**Security**
+* Stop a container runtime from executing CNI plugins
+* Restrict what a plugin does
+* Manage the security of the container it interacts with
+
+## Self-assessment use
+
+This self-assessment is created for the CNI project to perform an internal analysis of the
+project's security.  It is not intended to provide a security audit of CNI, or
```
Comment (ragashreeshekar) [inline assessments/projects/cni/self-assessment.md:130]:
```suggestion
This self-assessment is not intended to provide a security audit of CNI, or
```

**Review by ragashreeshekar — APPROVED** [review]
